cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/EndpointLoader.kt

```diff
@@ -0,0 +1,51 @@
+package com.cognifide.cogboard.config
+
+import io.vertx.core.json.JsonArray
+import io.vertx.core.json.JsonObject
+import java.util.*
+
+class EndpointLoader(private val endpointsConfig: JsonArray, private val credentialsConfig: JsonArray, private val endpointId: String) {
+
+    fun asJson(withSensitiveData: Boolean): JsonObject {
+        val endpoint = findEndpointById(endpointId, endpointsConfig)
+        if (withSensitiveData) endpoint.map { attachUserPassword(it) }
+        return endpoint.orElse(JsonObject())
+    }
+
+    private fun attachUserPassword(endpoint: JsonObject): JsonObject {
+        endpoint.remove(CREDENTIALS)?.let {
+            val credentials = findEndpointById(it as String, credentialsConfig)
```

Comment:
rename `findEndpointById` to smth more generic like `findById`

Reply:
Done and changed method into extension function for better reusability